tests/unittests/config/test_cc_ubuntu_advantage.py

```diff
@@ -197,160 +210,218 @@ def test_schema_validation(self, config, error_msg):
+            pytest.param(
+                {
+                    "ubuntu-advantage": {"token": "nope", "enable": ["wrong"]},
+                    "ubuntu_advantage": {"token": "token", "enable": ["esm"]},
+                },
+                dict(cloud=None, log=None, args=None),
+                [
+                    (
+                        MPATH,
+                        logging.WARNING,
+                        'Deprecated configuration key "ubuntu-advantage"'
+                        " provided. Expected underscore delimited "
+                        '"ubuntu_advantage"; will attempt to continue.',
+                    )
+                ],
+                None,
+                [mock.call(token="token", enable=["esm"])],
+                id="prefers_new_style_config",
+            ),
+        ],
+    )
+    @mock.patch(f"{MPATH}.configure_ua")
+    @mock.patch(f"{MPATH}.maybe_install_ua_tools")
+    def test_handle(
+        self,
+        m_maybe_install_ua_tools,
+        m_configure_ua,
+        cfg,
+        handle_kwargs,
+        log_record_tuples,
+        maybe_install_call_args_list,
+        configure_ua_call_args_list,
+        caplog,
     ):
-        """All ubuntu_advantage config keys are passed to configure_ua."""
-        cfg = {"ubuntu_advantage": {"token": "token", "enable": ["esm"]}}
-        handle("nomatter", cfg=cfg, cloud=None, log=self.logger, args=None)
-        m_configure_ua.assert_called_once_with(token="token", enable=["esm"])
+        handle("nomatter", cfg=cfg, **handle_kwargs)
+        for record_tuple in log_record_tuples:
+            assert record_tuple in caplog.record_tuples
+        if maybe_install_call_args_list is not None:
+            assert (
+                maybe_install_call_args_list
+                == m_maybe_install_ua_tools.call_args_list
+            )
+        if configure_ua_call_args_list is not None:
+            assert configure_ua_call_args_list == m_configure_ua.call_args_list
 
-    @mock.patch("%s.maybe_install_ua_tools" % MPATH, mock.MagicMock())
-    @mock.patch("%s.configure_ua" % MPATH)
-    def test_handle_warns_on_deprecated_ubuntu_advantage_key_w_config(
-        self, m_configure_ua
+    @pytest.mark.parametrize(
+        "cfg, handle_kwargs, match",
+        [
+            pytest.param(
+                {"ubuntu-advantage": {"commands": "nogo"}},
+                dict(cloud=None, args=None),
+                (
+                    'Deprecated configuration "ubuntu-advantage: commands" '
+                    'provided. Expected "token"'
+                ),
+                id="key_dashed",
+            ),
+            pytest.param(
+                {"ubuntu_advantage": {"commands": "nogo"}},
+                dict(cloud=None, args=None),
+                (
+                    'Deprecated configuration "ubuntu-advantage: commands" '
+                    'provided. Expected "token"'
+                ),
+                id="key_dashed",
```

Comment:
non-unique ID doesn't really cause an error as pytest will call the tests key_dashed0 key_dashed1.

```suggestion
                id="key_underscore",
```

Reply:
Ups, my bad here. Fixed.